### DISCUSSION TITLE
Select Combobox

### DISCUSSION THREAD

**Discussion by jeffreylauwers** [nl-design-system/backlog#340] [Component suggestions]
<img width="480" alt="visuele weergave van een select combobox component" src="https://github.com/user-attachments/assets/8fb77721-f2b0-4bb7-9071-fda33ba9cd0b">

We willen graag meer leren over de Select Combobox component. In welke vorm, of vormen, wordt deze momenteel ingezet en toegepast binnen jouw organisatie?

Deel hieronder een link naar Figma, Storybook, Github of screenshots. Met deze informatie kunnen we bewijzen dat de component onderdeel moet worden van de NL Design System component library.

Daarnaast wordt het delen van onderzoek gewaardeerd. Dus heb je zelf onderzoek gedaan en zijn er inzichten voor dit component? Dan horen we dit ook graag!

[Link naar bericht in Slack]({link-naar-Slack-bericht})

---

## Naam

Select Combobox

## Doel

Biedt de mogelijkheid om één optie te selecteren, waarbij de getoonde opties kunnen worden gefilterd door tekst in een invoerveld in te voeren.

## Ook bekend als

Autocomplete, Autosuggest, Typeahead.

## Gerelateerde componenten

[Select](https://github.com/orgs/nl-design-system/discussions/239), Search Input

## Onderzoek

[Editable Combobox With List Autocomplete Example -  | APG | WAI | W3C](https://www.w3.org/WAI/ARIA/apg/patterns/combobox/examples/combobox-autocomplete-list/)
In dit artikel wordt o.a. informatie gegeven over Accessibility Features en Keyboard support voor een component zoals Select combobox.

[Building an accessible autocomplete control - Adam Silver](https://adamsilver.io/blog/building-an-accessible-autocomplete-control/)
In dit artikel beschrijf Adam Silver hoe hij een toegankelijk autocomplete formcontrol zou opbouwen.

[ARIA: combobox role - MDN docs](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/combobox_role)
In dit artikel wordt het gebruik/toepassing van de aria `combobox` rol beschreven.

[Combobox - Adobe Spectrum](https://spectrum.adobe.com/page/combo-box/)
Het Adobe Spectrum design system heeft een documentatiepagina voor de Combobox component. Zij beschrijven hier gedrag en richtlijnen voor gebruik met visuele ondersteuning.

**Comment by jeffreylauwers** [comment]
## Den Haag

[Figma - Huidig](https://www.figma.com/design/JpoY3waVoQGlLQzQXTL9nn/%F0%9F%8C%BB-Den-Haag-Design-System?node-id=10818-68987&t=NnWjLoPsu9u8ArgD-4)

---

### Screenshots / Varianten

**Figma huidig**

One option possible

![Scherm­afbeelding 2024-09-13 om 10 08 59](https://github.com/user-attachments/assets/b76146bb-ccb6-4aa9-aa72-f996b7b73b03)

Multiple options possible

![Scherm­afbeelding 2024-09-13 om 10 09 19](https://github.com/user-attachments/assets/6a848461-d3ef-4657-b27f-fe97c38ab857)

---

### Notities

- Onder de naam: Combo box.
- Documentatie aanwezig (Figma huidig)
- Single select en Mulit select variant
- Tekst invullen gebeurd direct onder de trigger en boven de lijst aan opties.
- Bij de Single select variant kan de gekozen optie worden verwijderd door op het 'cross' icoon te klikken.
- Bij de Multi select variant worden gekozen opties als 'Pill' weergegeven, deze kunnen worden verwijderd door op het 'cross' icoon binnen de Pill te klikken.

**Comment by jeffreylauwers** [comment]
## DUO

[Documentatie website](https://uno.dfront.rijkscloud.nl/#/components/combobox)

---

### Screenshots / Varianten

**Documentatie website**

Default

Default
![Scherm­afbeelding 2024-09-13 om 10 31 14](https://github.com/user-attachments/assets/00aab5a0-229d-4811-b652-592646d03c43)

Open
![Scherm­afbeelding 2024-09-13 om 10 32 02](https://github.com/user-attachments/assets/bf8a1eff-4162-4137-b316-2f0e0efd1a26)

Typing
![Scherm­afbeelding 2024-09-13 om 10 32 11](https://github.com/user-attachments/assets/6696d31a-0826-4c16-b991-e96d9ab51767)

Selected
![Scherm­afbeelding 2024-09-13 om 10 32 16](https://github.com/user-attachments/assets/bcc93c5e-3ac9-4c8c-a00f-1710a50e111e)

Met checkboxes

Default

![Scherm­afbeelding 2024-09-13 om 10 32 24](https://github.com/user-attachments/assets/a43e8695-d44d-41f0-8a6b-d6fbb057e4f8)

Open
![Scherm­afbeelding 2024-09-13 om 10 32 35](https://github.com/user-attachments/assets/5d008ef4-5ea1-40e1-b815-e38216dc759c)

Selected
![Scherm­afbeelding 2024-09-13 om 10 32 44](https://github.com/user-attachments/assets/2eba4935-c458-4526-8913-9de30112f292)

---

### Notities

- Onder de naam: Combo box.
- Documentatie aanwezig.
- Single select en Mulit select variant.
- 'Search' icon vooraan.
- Tekst invullen gebeurd binnen het veld.
- Wanneer er onder de component geen ruimte is binnen de viewport klapt de lijst met opties naar boven open.
- Bij de Single select variant kan de gekozen optie worden verwijderd door op het 'cross' icoon te klikken.
- Bij de Multi select variant worden gekozen opties als 'Chip' weergegeven, deze kunnen worden verwijderd door op het 'cross' icoon binnen de Chip te klikken. Wanneer er meer dan 2 opties zijn gekozen is ook een 'Verwijder alles' optie aanwezig.

**Reply by tomootes** [reply]
Wat ik goed vind aan deze versie is dat het helder is dat je kan zoeken. Bij veel van dit soort componenten ben je verrast dat er opties tevoorschijn komen op het moment dat je iets typt in het veld. Vaak lijkt het gewoon een leeg textveld. Hier zorgen het icoontje en hier driehoekje (carret) voor duidelijkheid.

**Comment by jeffreylauwers** [comment]
## Utrecht

- [Storybook](https://nl-design-system.github.io/utrecht/storybook/?path=/docs/css_css-combobox--docs)
- [GitHub](https://github.com/nl-design-system/utrecht/tree/main/components/combobox)

---

### Screenshots / Varianten

**Storybook**

Default
![Scherm­afbeelding 2024-09-13 om 11 15 55](https://github.com/user-attachments/assets/d7db14e4-1314-49e7-92bc-9fc5805b3e59)

Open
![Scherm­afbeelding 2024-09-13 om 11 16 03](https://github.com/user-attachments/assets/f60d167e-a030-4dde-8450-fbd8f8e42ca9)

Selected
![Scherm­afbeelding 2024-09-13 om 11 16 34](https://github.com/user-attachments/assets/4881cb18-a997-4651-a418-d5a0b1895434)

Color options
![Scherm­afbeelding 2024-09-13 om 11 16 56](https://github.com/user-attachments/assets/4229388d-809a-4e9e-98bd-2f06ab3f1e86)

Icon options
![Scherm­afbeelding 2024-09-13 om 11 17 01](https://github.com/user-attachments/assets/1676ce1c-4b4c-438d-98a5-2f5fb84251d7)

Font familiy options
![Scherm­afbeelding 2024-09-13 om 11 17 08](https://github.com/user-attachments/assets/802ac8d2-efe4-4d7f-a9a9-242b8a9f3488)

Right-to-left
![Scherm­afbeelding 2024-09-13 om 11 17 13](https://github.com/user-attachments/assets/20bf296b-8f7f-4d77-911f-47a5bd85fe06)

Long text
![Scherm­afbeelding 2024-09-13 om 11 17 20](https://github.com/user-attachments/assets/9c4e1951-de6c-427a-9845-aac36abb1459)

---

### Notities

- Onder de naam: Combobox.
- Documentatie aanwezig rondom toetsenbord interactie, role, aria en auto-complete.
- Geen iconen binnen veld.
- Veel verschillende varianten zoals bijvoorbeeld kleur, iconen en font-familiy
- Support voor right-to-left en lange teksten.

**Comment by jeffreylauwers** [comment]
## ✨ Dit component is nu Help Wanted ✨

Help je mee hem door de Community stappen te krijgen?
[Select Combobox op het Community bord](https://github.com/orgs/nl-design-system/projects/29/views/1?filterQuery=select+combobox)

**Comment by jiromaykin** [comment]
# Open Inwoner

[Storybook](https://maykinmedia.github.io/open-inwoner/?path=/docs/components-kvkbranchselector--docs)
[Github HTML structuur](https://github.com/maykinmedia/open-inwoner/tree/develop/src/open_inwoner/react/components/KVKBranchSelector) (Preact)
[Github design-tokens ](https://github.com/maykinmedia/open-inwoner-design-tokens/blob/main/src/components/combobox.tokens.json)

---

In open-inwoner hebben wij een design dat qua anatomie afwijkt: met een extra knop achter het zoekveld die de 'select' kan sluiten,
ook hebben we een 'check' icoon dat verschijnt zodra de selectie definitief is.
En een border.

Qua code is dit een utrecht-combobox die sterk uitelkaar getrokken is maar waar nog wel de utrecht-form-field etc. in zit.

---

<img width="675" height="573" alt="Screenshot 2026-04-17 at 15 33 46" src="https://github.com/user-attachments/assets/4b79f32d-d340-4f67-a22d-2198f5a93624" />

**Reply by jiromaykin** [reply]
Inmiddels is dit component in productie, dus bovenstaande tekst geüpdate.

**Comment by sergei-maertens** [comment]
In Open Formulieren hebben we hiervoor ook een implementatie op basis van utrecht community components en de react-select NPM package: https://github.com/open-formulieren/formio-renderer/tree/daa0545aedb48f56c1a9830de62fde3c3e4d23b0/src/components/forms/Select
